src/charm.py

```diff
@@ -165,13 +178,13 @@ def _render_conf_file(self) -> str:
         rendered = template.render(
             mappers_path=str(self._mappers_dir_path),
             identity_schema_file_path=self._identity_schema_file_path,
-            default_browser_return_url=self.config.get("login_ui_url"),
+            default_browser_return_url=self._get_login_ui_endpoint_info("default_url"),
             public_base_url=self._domain_url,
-            login_ui_url=join(self.config.get("login_ui_url"), "login"),
-            error_ui_url=join(self.config.get("login_ui_url"), "oidc_error"),
+            login_ui_url=self._get_login_ui_endpoint_info("login_url"),
```

Comment:
Just a suggestion: 
`self._get_login_ui_endpoint_info` could return an object that has all the endpoints. And then extract the necessary info in the jinja template. 

This way we could be consistent in how we handle data from relation interfaces (that's how we do it for `_get_database_relation_info` and `_get_hydra_endpoint_info`)